Automattic/jetpack
# Social Menu & Social Media Icons: Add X support

## Proposed changes:

Update the icon font, the social icons widget, the social menu, the social-logos package to include the new X icon.

Noting that this purposefully does not replace the existing Twitter icon. This will allow site owners to decide whether the icon they'd like to use. It should be less forceful that way, we will not update the existing Twitter icon that may be in use on many sites; site owners will do that themselves if they'd like. This is the approach that will most likely be taken for Core as well: https://github.com/WordPress/gutenberg/pull/54092

Related PR: https://github.com/Automattic/social-logos/pull/126/

<img width="553" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/edb59a2c-a30d-45e2-9850-aa70ec389b29">
<img width="480" alt="image" src="https://github.com/Automattic/jetpack/assets/426388/5456359b-9cf8-446c-965c-5ae0225b2ffa">


### Other information:


## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Go to `yoursite.com/wp-content/plugins/jetpack/_inc/social-logos/social-logos.html`
* Check the new logos
* Go to Jetpack > Settings and enable extra sidebar widgets.
* Install and activate the Dara theme on your site.
* Go to Plugins > Add New and install the Classic Widgets plugin.
* Go to Appearance > Customize > Menus
* Create a new menu with some links, e.g. `https://www.facebook.com/jetpackme` and `https://x.com/jetpack`
* Assign that menu to the Social Menu location
* Go to Appearance > Widgets menu
* Drag the Social Icons widget to one of the active widget areas, and add some links again.
* Check how things look like on the frontend, for both your social menu and the Social Icons Widget you created.
* Go to Jetpack > Settings > Social and enable sharing buttons. **Do not look for a X icon just yet, we are just checking to ensure there are no regressions with the existing icons**.
* Go to Settings > Sharing and add some sharing buttons to your site. Use the icon-only sharing type.
* The buttons should look good in the admin preview, as well as on the frontend when your changes are saved.
